p2panda-net/src/lib.rs

```diff
@@ -8,15 +8,36 @@ pub mod network;
 mod protocols;
 mod sync;
 
+use std::fmt::{Debug, Display};
+
 pub use addrs::{NodeAddress, RelayUrl};
 pub use config::Config;
 pub use message::{FromBytes, ToBytes};
 pub use network::{Network, NetworkBuilder, RelayMode};
+use p2panda_core::Hash;
 pub use protocols::ProtocolHandler;
 pub use tokio_util::task::AbortOnDropHandle;
 
 #[cfg(feature = "log-sync")]
 pub use p2panda_sync::log_sync::LogSyncProtocol;
 
 pub type NetworkId = [u8; 32];
-pub type TopicId = [u8; 32];
+
+/// Topics are identified by a network-unique id encoded as a `[u8; 32]`.
+/// Topic ids are announced on the network and used to identify peers with
+/// similar interests. Once identified, peers join a gossip overlay and, if
+/// a sync protocol has been provided, attempt to synchronize past state.
+///
+/// The `Topic` trait must be implemented on any user defined topic types.
+pub trait Topic: Clone + Debug + Display + Send + Sync {
```

Comment:
Do we need this to be `StdHash` (see other comment) and `Serialize` and `Deserialize` as well? If it's sent over the wire we want it to be serializable, and hashable to insert it into all sorts of hash maps etc.

Reply:
A `Topic` is actually never sent over the wire in `p2panda-net`, and per my other comment there is a separation between bounds required on topics in `-net` and `-sync`. Maybe this is confusing though? I thought the separation was good to make it clear that the id requirement was about gossip only, but if it causes confusion in other directions I'm happy to consider using the trait in `-sync` as well and adding more bounds.

Reply:
Ja, I got confused here, might be easier if they are somehow consolidated, will add a comment with some thoughts